ext/fetch/lib.rs

```diff
@@ -237,11 +237,11 @@ pub fn op_fetch<FP>(
 where
   FP: FetchPermissions + 'static,
 {
-  let client = if let Some(rid) = client_rid {
+  let (client, allow_host) = if let Some(rid) = client_rid {
```

Comment:
**comment:** This is the first time I see this sort of combination of `let` and `if let Some(X) = Y`. I am both in awe and appalled :D But, it makes sense after a while and was here before we got here so of course why not.